src/snowflake/cli/api/entities/application_package_entity.py

```diff
@@ -1,10 +1,33 @@
+from pathlib import Path
+
+from snowflake.cli._plugins.nativeapp.artifacts import build_bundle
+from snowflake.cli._plugins.nativeapp.bundle_context import BundleContext
+from snowflake.cli._plugins.nativeapp.codegen.compiler import NativeAppCompiler
+from snowflake.cli._plugins.workspace.action_context import ActionContext
 from snowflake.cli.api.entities.common import EntityBase
+from snowflake.cli.api.project.schemas.entities.application_package_entity_model import (
+    ApplicationPackageEntityModel,
+)
 
 
-class ApplicationPackageEntity(EntityBase):
+class ApplicationPackageEntity(EntityBase[ApplicationPackageEntityModel]):
     """
     A Native App application package.
     """
 
-    def bundle(self):
-        raise NotImplementedError("Application package bundle")
+    def bundle(self, ctx: ActionContext):
```

Comment:
Should we consider `action_bundle` so when we look for action we look for `action_{action}` method? It should make it easier to differentiate between helpers and action method until we have decorators. WDYT?

Reply:
Sure. Changed the enum value to `BUNDLE = "action_bundle"`